src/server/service/kv.rs

```diff
@@ -778,6 +899,46 @@ impl<T: RaftStoreRouter + 'static> tikvpb_grpc::Tikv for Service<T> {
         ctx.spawn(future);
     }
 
+    fn raw_batch_delete(
+        &self,
+        ctx: RpcContext,
+        mut req: RawBatchDeleteRequest,
+        sink: UnarySink<RawBatchDeleteResponse>,
+    ) {
+        const LABEL: &str = "raw_batch_delete";
+        let timer = self.metrics.raw_batch_delete.start_coarse_timer();
+
+        let keys = req.take_keys().into_iter().map(|x| x).collect();
+
```

Comment:
Ditto. BTW, please keep the same order of take keys and generate `future_callbacks` with `batch_put`.